CMakeLists.txt

```diff
@@ -579,12 +579,32 @@ if(UMF_USE_MSAN)
                     "prevent reporting false-positives")
     add_sanitizer_flag(memory)
 endif()
+
 # Fuzzer instrumentation for the whole library
 if(UMF_BUILD_FUZZTESTS
    AND CMAKE_CXX_COMPILER_ID MATCHES "Clang"
    AND LINUX)
     add_compile_options("-fsanitize=fuzzer-no-link")
     add_link_options("-fsanitize=fuzzer-no-link")
+
+    # We need to find the fuzzer lib in the LLVM installation dir and link it
+    # statically
+    string(REGEX MATCH "^[0-9]+" CMAKE_C_COMPILER_MAJOR_VERSION
+                 ${CMAKE_C_COMPILER_VERSION})
+    find_library(
+        FUZZER_NO_MAIN_LIB
+        NAMES libclang_rt.fuzzer_no_main-x86_64.a
+        PATHS
+            "/usr/lib/llvm-${CMAKE_C_COMPILER_MAJOR_VERSION}/lib/clang/${CMAKE_C_COMPILER_MAJOR_VERSION}/lib/linux/"
```

Comment:
could we use CMAKE_PREFIX_PATH or some custom var to include here - then in case of error we could print a msg like - "use \<var\> to set up the path for the clang libs..."

I can imagine that this hardcoded "/usr/lib/llvm-${CMAKE_C_COMPILER_MAJOR_VERSION}/lib..." path will not work on some distros...

Reply:
thanks, done. Additionally I updated the README